etc/cdn-log-shipper/log-shipper.yml

```diff
@@ -92,14 +92,15 @@ Resources:
         const crypto = require('crypto');
 
         const IPV4_MASK = /\.[0-9]{1,3}$/;
-        const IPV6_MASK = /:[0-9a-fA-F]{0,4}$/;
         const maskIp = (ip, field) => {
           if (ip.match(IPV4_MASK)) {
             return ip.replace(IPV4_MASK, '.0');
-          } else if (ip.endsWith(':')) {
-            return ip;
-          } else if (ip.match(IPV6_MASK)) {
-            return ip.replace(IPV6_MASK, ':0');
+          } else if (ip.includes(':')) {
+            // take up to 4 chunks - drop the rest
+            const chunks = ip.split(':').slice(0, 4);
```

Comment:
It seemed like that old regex didn't _really_ work for all cases.

This one is better, but I suspect it doesn't handle some edge cases where there are `::` in the middle of the string somewhere.

Reply:
So your example expands to `1234:5678:0:0:0:0:abcd:efgh`.  Ideally we'd truncate that to `1234:5678::` or equivalent.

Looks like my code would end up with 2 `::` in there - `1234:5678::abcd::`.  Which isn't valid (it's ambiguous how many 0s go where).

In reality, I'm doubtful we'd ever see that?  As zeroes tend to be towards the end of the IPv6.  Trying to think of how to handle without an actual IP parsing library...

Reply:
I think this is good enough, and if we need to bring in a lib, well, then we will!